feat(warnings): add `encryptedWarning` text parameter (#29120)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

Information provided mainly in debug log level.

> ## encryptedWarning

Reviewer: Here's the wishlish from `@rarkins`, as todos:

- [ ] A new page on `docs.renovatebot.com` explaining our preference for UI-uploaded secrets, and a section on how to migrate
- [ ] A warning message printed once per repo/job, which says something like "Support for encrypted secrets in config is deprecated by the Mend Renovate App" plus a link to this documentation page
- [ ] The warning is shown on Dependency Dashboard, and an indicator is shown in each PR mentioning that warnings were raised (not sure if we need the full message or not)

Source:

- https://github.com/renovatebot/renovate/discussions/28895#discussioncomment-9353343

Author: 1. we would like first to support migrating secrets in production before we add it to the docs.
2. the message itself will be set on our Mend App with the doc link.
3. i will probably add the PR warnings to this PR soon, i wasn't sure if we need it or not.

Author: i wrote a comment in the discussion asking how to do this from product perspective, resolving this comment.

> By default, if there's an encrypted secret in the logs no warning will appear.
> By setting `encryptedWarning` text, Renovate will print the warning to the logs and will show it in the dependency dashboard.

Reviewer: ```suggestion
By default, Renovate does _not_ warn you if it finds an encrypted secret in the logs.

But if you set an `encryptedWarning` text, then Renovate will:

- print the `encryptedWarning` text in the logs, and
- show the `encryptedWarning` text on your Dependency Dashboard
```

Information provided mainly in debug log level.

## encryptedWarning

> By default, Renovate does _not_ warn you if it finds an encrypted secret in your config.
> But if you set an `encryptedWarning` text, then Renovate will:
> 
> - print the `encryptedWarning` text in the logs
> - show the `encryptedWarning` text on your Dependency Dashboard
> - raise a warning in every pull request description

Reviewer 2: ```suggestion
Use this if you want to stop supporting `encrypted` configuration capabilities but want to warn users first to migrate.
If set to a string value, Renovate will:

- log warnings with the `encryptedWarning` text
- show the `encryptedWarning` text on the Dependency Dashboard of any repo using `encrypted` config
- include the text as a warning in every pull request
```

Reviewer: Looks good to me, but can you change two things?

- Expand `repo` to `repository`
- capitalize `Pull Request`

Information provided mainly in debug log level.

## encryptedWarning

Use this if you want to stop supporting `encrypted` configuration capabilities but want to warn users first to migrate.

> If set to a string value, Renovate will:
> 
> - log warnings with the `encryptedWarning` text
> - show the `encryptedWarning` text on the Dependency Dashboard of any repository using `encrypted` config
> - include the text as a warning in every Pull Request

Reviewer 2: ```suggestion
If set to a string value, Renovate will log warnings with the `encryptedWarning` text, meaning the message will be visible to users such as on the Dependency Dashboard.
```
